include/Core/Metas/QueryOverriders/DateQueryOverrider.php

```diff
@@ -0,0 +1,171 @@
+<?php
+
+namespace BulkWP\BulkDelete\Core\Metas\QueryOverriders;
+
+use BulkWP\BulkDelete\Core\Base\BaseQueryOverrider;
+
+/**
+ * Class that encapsulates the logic for handling date format specified by the user.
+ *
+ * @package Bulk Delete
+ *
+ * @author  Sudar
+ *
+ * @since   6.0
+ */
+class DateQueryOverrider extends BaseQueryOverrider {
+	/**
+	 * Delete Options.
+	 *
+	 * @var array
+	 */
+	protected $delete_options;
+	/**
+	 * Date format of meta value that is stored in `wp_post_meta` table..
+	 *
+	 * @var string
+	 */
+	protected $meta_value_date_format;
+
+	/**
+	 * Date format of input value that will be compared with meta value.
+	 *
+	 * @since 1.0
+	 *
+	 * @var string
+	 */
+	protected $input_value_date_format;
+
+	/**
+	 * Setup hooks and load.
+	 *
+	 * @since 1.0
+	 */
+	public function load() {
+		add_action( 'parse_query', array( $this, 'parse_query' ) );
+	}
+
+	/**
+	 * Creates query object after processing date with specified date format.
+	 *
+	 * @param array $delete_options Delete Options.
+	 *
+	 * @return \WP_Query $query Query object.
+	 */
+	public function get_query( $delete_options ) {
+		$query = $this->process_date_fields( $delete_options );
+
+		return $query;
+	}
+
+	public function process_date_fields( $delete_options ) {
+		if ( ! empty( $delete_options['relative_date'] ) && 'custom' !== $delete_options['relative_date'] ) {
+			$delete_options['meta_value'] = date( 'c', strtotime( $delete_options['relative_date'] ) );
+		}
+
+		if ( ! empty( $delete_options['date_unit'] ) && ! empty( $delete_options['date_type'] ) ) {
+			$interval_unit = $delete_options['date_unit'];
+			$interval_type = $delete_options['date_type'];
+
+			switch ( $delete_options['meta_op'] ) {
+				case '<':
+				case '<=':
+					$delete_options['meta_value'] = date( 'Y-m-d', strtotime( '-' . $interval_unit . ' ' . $interval_type ) );
+					break;
+				default:
+					$delete_options['meta_value'] = date( 'Y-m-d', strtotime( $interval_unit . ' ' . $interval_type ) );
+			}
+		}
+
+		// In v1.0 `date_format` was changed to `meta_value_date_format`.
+		// Needed?
+		if ( isset( $delete_options['date_format'] ) ) {
+			$delete_options['meta_value_date_format'] = $delete_options['date_format'];
+		}
+		$meta_query = array(
+			'key'     => $delete_options['meta_key'],
+			'value'   => $delete_options['meta_value'],
+			'compare' => $delete_options['meta_op'],
+			'type'    => $delete_options['meta_type'],
+		);
+
+		$options = array(
+			'meta_query' => array( $meta_query ),
+		);
+
+		if ( 'DATE' === $meta_query['type'] && ! empty( $delete_options['meta_value_date_format'] ) ) {
+			$options['cf_meta_value_date_format'] = $delete_options['meta_value_date_format'];
+
+			if ( ! empty( $delete_options['input_value_date_format'] ) ) {
+				$options['cf_input_value_date_format'] = $delete_options['input_value_date_format'];
+			} else {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method process_date_fields uses an else expression. Else is never necessary and you can simplify the code to work without else.](https://app.codacy.com/app/BulkWP/bulk-delete/pullRequest?prid=2562677)